rskj-core/src/test/java/co/rsk/peg/BridgeSupportGetTransactionTypeTest.java

```diff
@@ -0,0 +1,414 @@
+            Instant.ofEpochMilli(1000L),
+            0L,
+            bridgeConstants.getBtcParams()
+        );
+        when(provider.getNewFederation()).thenReturn(activeFederation);
+
+        Address userAddress = new Address(
+            btcParams,
+            Hex.decode("4a22c3c4cbb31e4d03b15550636762bda0baf85a")
+        );
+
+        // Create a tx from the Fed to a random btc address
+        BtcTransaction pegoutBtcTx = new BtcTransaction(btcParams);
+        pegoutBtcTx.addOutput(Coin.COIN, userAddress);
+        TransactionInput pegoutInput = new TransactionInput(
+            btcParams,
+            pegoutBtcTx,
+            new byte[]{},
+            new TransactionOutPoint(
+                btcParams,
+                0,
+                Sha256Hash.ZERO_HASH
+            )
+        );
+        pegoutBtcTx.addInput(pegoutInput);
+
+        FederationTestUtils.addSignatures(activeFederation, fedKeys, pegoutBtcTx);
+
+        // Act
+        PegTxType resultTxType = bridgeSupport.getTransactionType(pegoutBtcTx);
+
+        //Assert
+        Assertions.assertEquals(PegTxType.PEGOUT_OR_MIGRATION, resultTxType);
+    }
+
+    @Test
+    void getTransactionType_migration_tx() {
+        // Arrange
+        long federationActivationAge = bridgeConstants.getFederationActivationAge(mock(ActivationConfig.ForBlock.class)) + 5L;
+        org.ethereum.core.Block rskCurrentBlock = new BlockGenerator().createBlock(federationActivationAge, 1);
+        BridgeStorageProvider provider = mock(BridgeStorageProvider.class);
+        BridgeSupport bridgeSupport = new BridgeSupportBuilder()
+            .withBridgeConstants(bridgeConstants)
+            .withProvider(provider)
+            .withExecutionBlock(rskCurrentBlock)
+            .build();
+
+        Federation activeFederation = FederationTestUtils.getFederation(100, 200, 300);
+        when(provider.getNewFederation()).thenReturn(activeFederation);
+
+        List<BtcECKey> retiringFedKeys = Arrays.asList(
+            BtcECKey.fromPrivate(Hex.decode("fa01")),
+            BtcECKey.fromPrivate(Hex.decode("fa02")),
+            BtcECKey.fromPrivate(Hex.decode("fa03"))
+        );
+        retiringFedKeys.sort(BtcECKey.PUBKEY_COMPARATOR);
+
+        Federation retiringFederation = new Federation(
+            FederationTestUtils.getFederationMembersWithBtcKeys(retiringFedKeys),
+            Instant.ofEpochMilli(1000L),
+            0L,
+            bridgeConstants.getBtcParams()
+        );
+        when(provider.getOldFederation()).thenReturn(retiringFederation);
+
+        BtcTransaction migrationTx = new BtcTransaction(btcParams);
+        migrationTx.addOutput(Coin.COIN, activeFederation.getAddress());
+        TransactionInput migrationTxInput = new TransactionInput(
+            btcParams,
+            null,
+            new byte[]{},
+            new TransactionOutPoint(
+                btcParams,
+                0,
+                Sha256Hash.ZERO_HASH
+            )
+        );
```

Comment:
Shouldn't This Input Come From The Retiring Federation?

Reply:
It comes. Below you can find where the signature retiring federation redeem script and signatures are added. See: `FederationTestUtils.addSignatures`.